pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/tenant/TenantRebalanceResult.java

```diff
@@ -65,4 +215,449 @@ public void setJobId(String jobId) {
+        Map<String, AggregatedServerSegmentChangeInfo> serverAggregates) {
+      super(0, 0, 0, 0, 0, null, null, null, null);
+
+      if (summaryResults.isEmpty()) {
+        return;
+      }
+
+      int totalSegmentsToBeMoved = 0;
+      int totalSegmentsToBeDeleted = 0;
+      long totalEstimatedDataToBeMovedInBytes = 0;
+
+      int beforeRebalanceReplicationFactor = 0;
+      int afterRebalanceReplicationFactor = 0;
+      int beforeRebalanceSegmentsInSingleReplica = 0;
+      int afterRebalanceSegmentsInSingleReplica = 0;
+      int beforeRebalanceSegmentsAcrossAllReplicas = 0;
+      int afterRebalanceSegmentsAcrossAllReplicas = 0;
+
+      // Track consuming segment summaries for realtime tables
+      List<RebalanceSummaryResult.ConsumingSegmentToBeMovedSummary> consumingSummaries = new ArrayList<>();
+      int validSummariesCount = 0;
+
+      // Aggregate data from individual table summaries
+      for (RebalanceSummaryResult summary : summaryResults) {
+        if (summary.getSegmentInfo() != null) {
+          RebalanceSummaryResult.SegmentInfo segmentInfo = summary.getSegmentInfo();
+          validSummariesCount++;
+
+          totalSegmentsToBeMoved += segmentInfo.getTotalSegmentsToBeMoved();
+          totalSegmentsToBeDeleted += segmentInfo.getTotalSegmentsToBeDeleted();
+          if (totalEstimatedDataToBeMovedInBytes >= 0) {
+            totalEstimatedDataToBeMovedInBytes = segmentInfo.getTotalEstimatedDataToBeMovedInBytes() < 0 ? -1
+                : totalEstimatedDataToBeMovedInBytes + segmentInfo.getTotalEstimatedDataToBeMovedInBytes();
+          }
+
+          if (segmentInfo.getNumSegmentsInSingleReplica() != null) {
+            beforeRebalanceSegmentsInSingleReplica +=
+                segmentInfo.getNumSegmentsInSingleReplica().getValueBeforeRebalance();
+            afterRebalanceSegmentsInSingleReplica +=
+                segmentInfo.getNumSegmentsInSingleReplica().getExpectedValueAfterRebalance();
+          }
+
+          if (segmentInfo.getNumSegmentsAcrossAllReplicas() != null) {
+            beforeRebalanceSegmentsAcrossAllReplicas +=
+                segmentInfo.getNumSegmentsAcrossAllReplicas().getValueBeforeRebalance();
+            afterRebalanceSegmentsAcrossAllReplicas +=
+                segmentInfo.getNumSegmentsAcrossAllReplicas().getExpectedValueAfterRebalance();
+          }
+
+          if (segmentInfo.getConsumingSegmentToBeMovedSummary() != null) {
+            consumingSummaries.add(segmentInfo.getConsumingSegmentToBeMovedSummary());
+          }
+        }
+      }
+
+      if (validSummariesCount == 0) {
+        return;
+      }
+
+      int maxSegmentsAddedToASingleServer = 0;
+      for (AggregatedServerSegmentChangeInfo aggregate : serverAggregates.values()) {
+        maxSegmentsAddedToASingleServer = Math.max(maxSegmentsAddedToASingleServer, aggregate.getSegmentsAdded());
+      }
+
+      // Calculate average segment size
```

Comment:
is this meaningful for a tenant rebalance? tables can have very different segment size characteristics. no harm in keeping it around though I guess